server/src/main/java/com/google/musicanalysis/api/youtube/YoutubeRequest.java

```
@@ -38,6 +39,12 @@ public String getResult() throws MalformedURLException, IOException {
 
     URL url = buildUrl();
 
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    int response = con.getResponseCode();
+    if (response > 400) {
+      return "unreadable";
```

throw an exception instead of returning this string